#!/usr/bin/env python

PACKAGE = "whycon"
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("identify",       bool_t,   0, "Whether to identify IDs",      True);
gen.add("circle_diameter", double_t, 0, "Marker's outer diameter [m]",  0.122, 0.0, 100.0);
gen.add("num_markers",     int_t,    0, "Number of markers to track",   5, 0, 1000);
gen.add("min_size",        int_t,    0, "Min size of marker [px]",      30, 10, 1000);

gen.add("field_length", double_t, 0, "Distance of markers in x-axis in custom coordinate system", 1.0, 0.0, 100.0);
gen.add("field_width",  double_t, 0, "Distance of markers in y-axis in custom coordinate system", 1.0, 0.0, 100.0);

gen.add("initial_circularity_tolerance", double_t, 0, "Initial circularity test tolerance [%]", 100.0, 0.0, 100.0)
gen.add("final_circularity_tolerance",   double_t, 0, "Final circularity test tolerance [%]",   2.0,  0.0, 100.0)

gen.add("area_ratio_tolerance",           double_t, 0, "Tolerance of black and white area ratios [%]", 40.0,  0.0, 200.0)
gen.add("center_distance_tolerance_ratio", double_t, 0, "Concentricity test ratio [%]",                 10.0,  0.0, 100.0)
gen.add("center_distance_tolerance_abs",   double_t, 0, "Concentricity test absolute [px]",              5.0,  0.0, 25.0)

exit(gen.generate(PACKAGE, "whycon", "whycon"))
